File: lc121.js
var maxProfit = (prices)=>{
    let n = prices.length;
    let mn = 999999;
    let ans = 0;
    for(let i=0;i<n;i++){
        mn = Math.min(mn, prices[i]);
        ans = Math.max(ans, prices[i]-mn);
    }
    return ans;
};

File: py/lc121.py
import sys

def maxProfit(prices):
  n = len(prices)
  mn = sys.maxsize
  ans = 0

  for i in range(n):
    if prices[i]<mn:
      mn = prices[i]
    if (prices[i]-mn)>ans:
      ans = prices[i]-mn
  
  return ans